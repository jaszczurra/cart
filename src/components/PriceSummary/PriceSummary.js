import React from 'react'
import PropTypes from 'prop-types'

import { convertPrice } from '../../utils/helpers'

const PriceSummary = ({ values }) => {
  const { products } = values
  const sum = products.reduce((acc, curr) => acc + curr.quantity * +curr.price, 0)
  const convertedSum = convertPrice(sum)

  return (
    <div className='price-summary'>{`Suma: ${convertedSum}z≈Ç`}</div>
  )
}

PriceSummary.propTypes = {
  values: PropTypes.object.isRequired
}

PriceSummary.defaultProps = {
  values: {}
}

export { PriceSummary }
