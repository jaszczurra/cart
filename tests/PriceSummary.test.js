import React from 'react'
import renderer from 'react-test-renderer'

import { PriceSummary } from '../src/components/PriceSummary/PriceSummary'

const mockedProps = {
  values: {
    products: [
      { price: 1, quantity: 3 },
      { price: 4.50, quantity: 1 },
      { price: 9.00, quantity: 2 },
    ]
  }
}

describe('PriceSummary', () => {

  it('Snapshot', () => {
    const component = renderer.create(<PriceSummary {...mockedProps} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
