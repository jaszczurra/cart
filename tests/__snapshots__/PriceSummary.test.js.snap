// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceSummary Snapshot 1`] = `
<div
  className="price-summary"
>
  Suma: 25,50z≈Ç
</div>
`;
